From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Sun, 26 Oct 2025 14:08:27 +0800
Subject: [PATCH] Rebased Paper to Tralux.


diff --git a/net/minecraft/server/packs/repository/ServerPacksSource.java b/net/minecraft/server/packs/repository/ServerPacksSource.java
index 5c1abc8ac942647c65b7f549065ae8e6a0ddab9e..31d176c4f1f1ade88dc391137373f49cf213d4b7 100644
--- a/net/minecraft/server/packs/repository/ServerPacksSource.java
+++ b/net/minecraft/server/packs/repository/ServerPacksSource.java
@@ -71,7 +71,7 @@ public class ServerPacksSource extends BuiltInPackSource {
         // Paper start - custom built-in pack
         final PackLocationInfo info;
         final PackSelectionConfig packConfig;
-        if ("paper".equals(id)) {
+        if ("paper".equals(id) || "tralux".equals(id)) {
             info = new PackLocationInfo(id, title, PackSource.BUILT_IN, Optional.empty());
             packConfig = new PackSelectionConfig(true, Pack.Position.TOP, true);
         } else {
