From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Sun, 9 Nov 2025 11:28:34 +0800
Subject: [PATCH] Blank Tralux Configuration System.


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index dd05a184919dd4737dd0a0194b83eadb1adc4699..33f39ab606752b12561ff820be1d5de62dff11b9 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -564,7 +564,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
         dataStorage.computeIfAbsent(ServerScoreboard.TYPE);
     }
 
-    protected abstract boolean initServer() throws IOException;
+    protected abstract boolean initServer() throws Exception;
 
     public ChunkLoadStatusView createChunkLoadStatusView(final int radius) {
         return new ChunkLoadStatusView() {
diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index 354b48c88b113c4048e258f2402aad2647d3c364..3a61d616a4847c9c97364439b3646aa1580eefd7 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -169,7 +169,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
     }
 
     @Override
-    public boolean initServer() throws IOException {
+    public boolean initServer() throws Exception {
         int i = this.getProperties().managementServerPort;
         if (this.getProperties().managementServerEnabled) {
             String string = this.settings.getProperties().managementServerSecret;
@@ -318,6 +318,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
             if (true) throw new IllegalStateException("Failed to bind to port", var11); // Paper - Propagate failed to bind to port error
             return false;
         }
+        me.coderfrish.tralux.core.TraluxBootstrap.setup();
 
         // CraftBukkit start
         this.server.loadPlugins();
